operator/builtin/input/aws/cloudwatch/cloudwatch_test.go

```diff
@@ -0,0 +1,284 @@
+package cloudwatch
+
+import (
+	"testing"
+	"time"
+
+	"github.com/observiq/stanza/operator/helper"
+	"github.com/observiq/stanza/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuild(t *testing.T) {
+	basicConfig := func() *CloudwatchInputConfig {
+		cfg := NewCloudwatchConfig("test_operator_id")
+		cfg.LogGroupName = "test"
+		cfg.Region = "test"
+		return cfg
+	}
+
+	test := "test"
+	var testStreams = []*string{&test}
+	cases := []struct {
+		name      string
+		input     *CloudwatchInputConfig
+		expectErr bool
+	}{
+		{
+			"default",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"log-stream-name-prefix",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.LogStreamNamePrefix = ""
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"event-limit",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.EventLimit = 5000
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"poll-interval",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.PollInterval = helper.Duration{Duration: 15 * time.Second}
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"profile",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.Profile = "test"
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"log-stream-names",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.LogStreamNames = testStreams
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"startat-end",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.StartAt = "end"
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"logStreamNames and logStreamNamePrefix both parameters Error",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.LogStreamNames = testStreams
+				cfg.LogStreamNamePrefix = "test"
+				return cfg
+			}(),
+			true,
+		},
+		{
+			"startat-beginning",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.StartAt = "beginning"
+				cfg.LogStreamNamePrefix = "test"
+				return cfg
+			}(),
+			false,
+		},
+		{
+			"poll-interval-invalid",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.PollInterval = helper.Duration{Duration: time.Second * 0}
+				return cfg
+			}(),
+			true,
+		},
+		{
+			"event-limit-invalid",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.EventLimit = 10001
+				return cfg
+			}(),
+			true,
+		},
+		{
+			"default-required-startat-invalid",
+			func() *CloudwatchInputConfig {
+				cfg := basicConfig()
+				cfg.StartAt = "invalid"
+				return cfg
+			}(),
+			true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg := tc.input
+			_, err := cfg.Build(testutil.NewBuildContext(t))
```

Comment:
A test like this should validate that the operator was built correctly. 

You should be able to define the expected state of the resulting operator for each config.